src/site/content/en/fast/publish-modern-javascript/index.md

```diff
@@ -16,5 +16,383 @@ codelabs:
+
+module.exports = {
+  //...
+  module: {
+    rules: [
+      // your existing babel-loader configuration:
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'babel-loader',
+          options: {
+            presets: [
+              [
+                '@babel/preset-env',
+                {
+                  targets: {
+                    browsers: ['last 2 versions', 'not dead'],
+                  },
+                },
+              ],
+            ],
+          },
+        },
+      },
+    ],
+  },
+  plugins: [new BabelEsmPlugin()],
+};
+```
+
+`BabelEsmPlugin` supports a wide array of webpack configurations, because it
+runs two largely separate builds of your application. Compiling twice can take a
+little bit of extra time for large applications, however this technique allows
+`BabelEsmPlugin` to integrate seamlessly into existing webpack configurations
+and makes it one of the most convenient options available.
+
+### webpack-plugin-modern-npm
+
+If you are using `babel-loader`, `webpack-plugin-modern-npm` is a plugin that
+can automatically compile all modern language features found in `node_modules`
+to ES2017 while allowing you to handle transpiling your first-party code
+separately. This doesn't generate modern and legacy bundles for a
+module/nomodule setup, but it does make it possible to install and use npm
+packages that contain modern JavaScript without breaking older browsers.
+
+```js
+const ModernNpmPlugin = require('webpack-plugin-modern-npm');
+
+module.exports = {
+  plugins: [
+    // auto-transpile modern stuff found in node_modules
+    new ModernNpmPlugin(),
+  ],
+};
+```
+
+When using this approach, you'll need to ensure modern syntax is supported by
+your minifier. Both [Terser](https://github.com/terser/terser#minify-options)
+and [uglify-es](https://github.com/mishoo/UglifyJS/tree/harmony#minify-options)
+have an option to specify `{ecma: 2017}` in order to preserve and in some cases
+generate ES2017 syntax during compression and formatting.
+
+## Rollup
+
+Rollup has built-in support for generating multiple sets of bundles as part of
+a single build, and generates modern code by default. As a result, Rollup can
+be configured to generate modern and legacy bundles with the official plugins
+you're likely already using.
+
+### @rollup/plugin-babel
+
+If you use Rollup, the
+[`getBabelOutputPlugin()` method](https://github.com/rollup/plugins/tree/master/packages/babel#running-babel-on-the-generated-code)
+provided by Rollup's
+[official Babel plugin](https://github.com/rollup/plugins/tree/master/packages/babel)
+transforms the code in generated bundles rather than individual source modules.
+Rollup has built-in support for generating multiple sets of bundles as part of
+a single build, each with their own plugins. You can use this to produce
+different bundles for modern and legacy by passing each through a different
+Babel output plugin configuration:
```

Comment:
```suggestion
If you use Rollup, the
[`getBabelOutputPlugin()` method](https://github.com/rollup/plugins/tree/master/packages/babel#running-babel-on-the-generated-code)
(provided by Rollup's
[official Babel plugin](https://github.com/rollup/plugins/tree/master/packages/babel))
transforms the code in generated bundles rather than individual source modules.
Rollup has built-in support for generating multiple sets of bundles as part of
a single build, each with their own plugins. You can use this to produce
different bundles for modern and legacy by passing each through a different
Babel output plugin configuration:
```